kernel/src/hil/accel.rs

```diff
@@ -0,0 +1,52 @@
+//! Interface for Accelerators
+
+use crate::common::leasable_buffer::LeasableBuffer;
+use crate::ErrorCode;
+
+/// Implement this trait and use `set_client()` in order to receive callbacks.
+pub trait Client<'a, const T: usize> {
+    /// This callback is called when the binary data has been loaded
+    /// On error or success `input` will contain a reference to the original
+    /// data supplied to `load_binary()`.
+    fn binary_load_done(&'a self, result: Result<(), ErrorCode>, input: &'static mut [u8]);
```

Comment:
This interface seems somewhat OTBN specific -- there are plenty of hardware accelerators that don't load a binary in order to be used.

Reply:
If no binary data is required then this doesn't have to be called. So I don't see that it would break non OT users